mobile_app_connector/controllers/mobile_app_controller.py

```diff
@@ -73,9 +73,13 @@ def get_hub_messages(self, partner_id, **parameters):
         """
         hub_obj = request.env['mobile.app.hub'].sudo()
         if partner_id:
-            # This will ensure the user is logged in
-            request.session.check_security()
-            hub_obj = hub_obj.sudo(request.session.uid)
+            # Check if requested url correspond to the current user
+            if partner_id == request.env.user.partner_id.id:
```

Comment:
Did you test it? Are you sure `request.env.user` is set at that point? I'm not sure that the variable exists here but I may be wrong.

Reply:
Yes it works, I already tested it and I just checked again

Reply:
Ok, then it looks good. However it's breaking the automated tests in `test_mobile_app_http`. Can you have a look at it? You can see the errors in the Travis build.